lib/eslint.js

```diff
@@ -657,12 +660,13 @@ module.exports = (function() {
      *      Mostly useful for testing purposes.
      * @returns {Object[]} The results as an array of messages or null if no messages.
      */
-    api.verify = function(text, config, filename, saveState) {
+    api.verify = function(textOrSourceCode, config, filename, saveState) {
 
         var ast,
             shebang,
             ecmaFeatures,
-            ecmaVersion;
+            ecmaVersion,
+            text = (typeof textOrSourceCode === "string") ? textOrSourceCode : null;
```

Comment:
Why not just do this check on line 679 and never have this variable available. So that we dont use it later in the code by mistake as it might produce wrong results.